### PR TITLE
[dem] [sdem] A bit of cleanup and update (Logger, Super, getattr)

### PR DESCRIPTION
**Pull request by maceligueta**
**Description**
Several pending issues are addressed with this PR

**Changelog**
- SwimmingDEM uses now he Kratos Logger instead of `print` or `KRATOS_WATCH` (#4540)
- One line with `getattr` has been migrated to` KratosGlobals.GetVariable` (#5495)
- Some Python2 style `super` have been updated to Python3

### CODE REVIEW

**Comment by maceligueta** [comment]
@philbucher : I could not replace the getattr in this line:

https://github.com/KratosMultiphysics/Kratos/blob/6d4eb17043de165502b69f93d1c599f843633f76/applications/SwimmingDEMApplication/python_scripts/swimming_sphere_strategy.py#L12

because it is not getting a `variable`, but an instance of a `scheme`. I tried, but it did not work.
It would be nice to clean this one and also the one in DEM, which is using `eval` (even worse!).

**Comment by philbucher** [comment]
@maceligueta in Structural and Fluid we have the [python-wrappers](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/python_solvers_wrapper_structural.py#L30-L50) where one "registers" the types by name

but probably a better solution will be the factories => #3185

**Comment by philbucher** [comment]
to me this PR looks good, nice cleanup thx @maceligueta !

File: applications/SwimmingDEMApplication/python_scripts/cellular_flow/run_pouliot_benchmark_2D.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
                 error = sys.exc_info()
                 combinations_that_failed.append({'size':size, 'type':derivatives_type})
 
-print()
+)
```
Comment (philbucher) [inline applications/SwimmingDEMApplication/python_scripts/cellular_flow/run_pouliot_benchmark_2D.py]:
```suggestion
```

**Review by GuillermoCasas — APPROVED** [review]
Nice job! Thanks for all that cleanup

**Pull request merged**